### ISSUE TITLE
add a way to see interactions/replies from non-bridged AP users

### ISSUE THREAD

**Issue by mackuba** [snarfed/bridgy-fed#1200]
Right now replies and likes from users who have enabled the bridge are mirrored on the other side, but replies and likes from users who haven't obviously can't be handled the same way. But I think it's still pretty important to be able to see them. This is easy on the Bluesky side (just open the post on bsky.app), but not in the other direction. So if you have a Bluesky account and you enable bridging to Mastodon, and some non-bridged Fediverse users reply to you, you have no way of seeing these replies easily. You actually don't know if maybe one of your posts went viral there and you got 20 comments, but you only know about 3 of them from users who are bridged. For me this is a pretty important missing piece if I wanted to "sunset" my Mastodon account and only post on Bluesky with a mirror via Bridgy, because I'd want to see what people are replying on Fedi.

@snarfed said that these responses should be showing up here: https://fed.brid.gy/bsky/mackuba.eu/notifications, but they don't seem to be. I've asked `@rademaker@mastodon.social` (who isn't following the bridge account) to make a few comments on my bridged Bluesky account, and these replies aren't shown anywhere, e.g.: https://mastodon.social/@rademaker/112814874617446470 (the Bridgy notifications list only shows that he followed me, but no further interactions).

**Comment by snarfed** [comment]
Thanks for filing! Yeah we started short circuiting out of processing earlier for non-opted-in accounts a while ago; sorry for misleading you here.

I'm a little cautious about this for non-opted-in accounts, but only a little, it's hopefully reasonable.

However, re:

> For me this is a pretty important missing piece if I wanted to "sunset" my Mastodon account and only post on Bluesky with a mirror via Bridgy, because I'd want to see what people are replying on Fedi.

Totally get it, that's an entirely reasonable desire, but I'm not sure if BF will be able to fully serve it. BF is only really designed to bridge interactions that are bridgeable, not to collect and serve all interactions. I can consider it, and look into it, but I'm not too optimistic just yet.

**Comment by mackuba** [comment]
> This is easy on the Bluesky side (just open the post on bsky.app), but not in the other direction

Actually, now that I think about it, it would also be useful for Mastodon accounts bridged to Bluesky, since this way you can easily check responses to a single post, but not to any of your posts in general.

**Comment by Tamschi** [comment]
Now that service DMs are implemented, would it be reasonable to provide (some) notifications that way?

If yes, then this probably should be an opt-in feature that the bridging-enabled- and help-messages mention.
It should also be possible to activate it for follows and replies independently of boosts and likes, since the latter could get spammy if supported.

**Comment by snarfed** [comment]
Lots more discussion on this in #1535

**Comment by Shanesan** [comment]
Hi, came over from #1535 as suggested by @snarfed where I proposed a way of DMing the Bridgy bot to reply to a post on your bridged account by linking to the reply and providing a response. 

The proposal returned the question: If interacting with unbridged replies who haven't opted into the bridge is that critical, make an account on that service, even if only to reply to these users. Which is fair, but kind of defeats the purpose of Bridgy; if you have said account already, you could use another service such as [Skymoth](https://skymoth.app/login) or [Sociabli](https://sociab.li/) or a simple script to automatically crosspost between BSky and Mastodon where there is no need for any user to opt in.

But Bridgy comes in with the assumption that you _don't_ have an account on the other side, so we are firing posts into the void without a way to respond if the replying user chooses not to Bridge their account, or opt in.

I'll say three things on this, and an important thing to note:

1. I don't believe a DM to you from Bridgy is overstepping the Opt In policy, but Snarfed does which is fine.
2. I do believe that Bridgy needs a way to continue the conversation even if the user does not Opt In since you can find said post, publicly, on the other side (usually) and see the responses to them.
3. Bridgy _could_ notify via DM that there are replies to a post (but not provide them verbatim, maybe with a link?) and see if you care to respond which technically doesn't violate any Opt In rules, or not notify you at all, but still give the option to respond by providing the URL to the reply and a message.

3 definitely promotes the idea that the "bridge" is two-way even if a user doesn't want to be part of the Bridge simply because they replied to a Bridged users' post.

Some people want to chat in the shallow water, not Bridge into the deep.

**Comment by snarfed** [comment]
Here's a rough design for a first pass at this:
* Start with read-only, one-way notifications of new, non-bridged replies (and maybe quotes; not likes or reposts) to your bridged posts.
* Sent via DM, batched hourly, per user. Includes links to the replies but no other content or profile info from them.
* May be turned off by DM response, eg `stop notifications`.
* Not sure if it would be default on or default off. I've heard from people on either side.

cc @anujahooja

**Comment by Shanesan** [comment]
> Here's a rough design for a first pass at this:
> 
>     * Start with read-only, one-way notifications of new, non-bridged replies, and maybe quotes, and maybe @-mentions, but not likes or reposts.
> 
>     * Sent via DM, batched hourly, per user. Includes links to the replies but no other content or profile info from them.
> 
>     * May be turned off by DM response, eg `stop notifications`.
> 
>     * Not sure if it would be default on or default off. I've heard from people on either side.

Love it. As for default on or default off, IMO as long as people know that the feature *exists* it doesn't need to be default on (for this first pass or beyond). Some people want to fire into the void, and that's fine. I'm simply thinking about the extra CPU cycles for people who may not care for the notifications.

**Comment by quillmatiq** [comment]
Hey all - we were starting to test a new Bridgy Fed feature where we send DMs for unbridged notifications, so that the user is aware of replies and mentions that aren't available on their native platform. Right now, it's configured to send an hourly digest if there are notifications to share. Let us know if you'd like to participate in the test and which account(s) you'd like to enable it for. 

@mackuba @Tamschi @Shanesan - tagging y'all since you were interested in this feature 🙂

**Comment by Shanesan** [comment]
Hi @anujahooja, I'm happy to participate in the test! shanie@tails.ch

**Comment by Tamschi** [comment]
I could test it on `@tamschi.itch.io` and `@Tamschi@mastodon.gamedev.place`, but my interaction volume is (most likely!) so low that it could take weeks before I get any notification this way.

**Comment by quillmatiq** [comment]
We'll add you anyway @Tamschi, just in case 🙂

I'll add you both shortly!

**Comment by lupomancer** [comment]
Hey @anujahooja if you guys are still looking for people to test this, I'd LOVE to participate in the testing.

This specific issue is a big blocker for my (admittedly very large) desire to use Bounce, so I'd like to see how this works practically

Thanks!

**Comment by mackuba** [comment]
Oh yeah, me too :)

**Comment by TedTed** [comment]
Happy to try this out as well!

**Comment by Baralheia** [comment]
If you're still accepting testers, I'd also be very interested in helping to test this new feature for my @baralheia@dragonchat.org account :)

**Comment by qazmlp** [comment]
I'll follow suit and say that this would be helpful on `@qazm@tiggi.es` and `@qazm.de` too. `@qazm@furries.club` and `@qazm@pixel.tiggi.es` maybe too, but those accounts are unlikely to have much bridged original content.

Do you pick up repost interaction notifications too or is that a different mechanism? If yes, it would be good to make those optional.

**Comment by quillmatiq** [comment]
Added y'all to the DM notifications test. Let me know if there are other accounts you want on it: https://github.com/snarfed/bridgy-fed/commit/e5918ab02552a29aaaf401e909ad2ec5163c16db

@lupomancer didn't have your profiles, so I haven't added you yet - let me know and I'll add them in 🙂

**Comment by lupomancer** [comment]
@anujahooja accounts are link removed and link removed

**Comment by quillmatiq** [comment]
@lupomancer added!

**Comment by Baralheia** [comment]
@anujahooja Hey thanks for adding me! About how long of a delay should I expect before the hourly digest is sent? It's been over two hours since I made a test post and haven't gotten anything from Bridgy despite multiple replies and favorites on the bridged post.

This is the bridged test post I mentioned: https://bsky.app/profile/baralheia.dragonchat.org/post/3lwctojtzma62

EDIT: It seems I may have just been impatient? I just got a DM from Bridgy for a *new* reply to my bridged post. I never got anything for the first few replies tho - maybe I posted before the change got pushed to prod?

**Comment by snarfed** [comment]
@Baralheia yup, this got deployed at roughly 00:28 UTC. Glad to hear you've gotten at least one notification since then!

**Comment by mackuba** [comment]
I also didn't get anything yet, but maybe just nobody likes me there ;) I replied now from an alt, will see if I get something. How often do the digests come?

**Comment by snarfed** [comment]
Hourly!

**Comment by mackuba** [comment]
Yess it worked! ✅

**Comment by qazmlp** [comment]
These notifications are supposed to come from e.g. `@ap.brid.gy` on Bluesky, right? If so, I think I figured out why it's not working even though I made a post that likely got unbridged interactions:

I have a list block against Jesse Singal's followers. The bridge account unfortunately still follows him too, so I don't receive Chat messages from it.

I can't unblock the bridge individually and I won't unsubscribe from that moderation list. Considering that following users back to confirm has been replaced by Chat, would you consider cleaning up those legacy follows? They most likely trip other automated mod lists like the "over …K follows" ones that people use to fight spam, too.

**Comment by snarfed** [comment]
Hah, funny. (Kind of. 😁) We only stopped @ap.brid.gy following back on Bluesky, we still need it on the fediverse, but yeah, this is a reasonable request. Good sleuthing. It's pretty distinct, mind filing a separate issue to track?

**Comment by mackuba** [comment]
Lol, even poor Bridgy Bot gets random ricochet listblocks too 🙃

**Comment by snarfed** [comment]
We're opening this up to everyone today! https://blog.anew.social/launch-dm-notifications/
